### PR TITLE
Removed all information about card values from HTML

### LINKED ISSUE #2: prevent winning by inspect element
currently, players can use inspect element and easily win the game by looking at the HTML. Implement some mechanism to prevent this

### PR DESCRIPTION
**Pull request by DNA5769**
Fixes #2 

Firstly I removed all information about the face of a card from the HTML. This would result each card's HTML going from this
```html
<!-- index.html -->
<div class="memory-card" data-character="inosuke">
  <img class="front-face" src="images/inosuke.png" alt="inosuke" />
  <img class="back-face" src="images/cover.png" alt="card cover" />
</div>
```
to this
```html
<!-- index.html -->
<div class="memory-card">
  <img class="front-face" src="#" alt="card front face" />
  <img class="back-face" src="images/cover.png" alt="card cover" />
</div>
```

Now I stored the srcs of each card in a list
```js
// index.js
const cardImageSrcs = [
    'images/cards/inosuke.png',
    'images/cards/nezuko.png',
    'images/cards/rengoku.png',
    'images/cards/mask.png',
    'images/cards/tanjiro.png',
    'images/cards/zenitsu.png',
    'images/cards/inosuke.png',
    'images/cards/nezuko.png',
    'images/cards/rengoku.png',
    'images/cards/mask.png',
    'images/cards/tanjiro.png',
    'images/cards/zenitsu.png',
];
```

So correspondily I changed the game logic as well i.e logic for flipping the card and check for matches. These are some of the important changes in both actions

Flipping the card
```js
// index.js
function flipCard(e, i){
    ...
    // This means that card is already flipped currently, so we exit out of function
    if (card.children[0].src!==window.location.href+"#")
        return;
    ...
    card.children[0].src=cardImageSrcs[i]; // Setting image source for flipped front face
    card.children[0].alt=cardImageSrcs[i].split('/').slice(-1)[0].slice(0, -4); // Setting image file name as alt text for flipped front face
    card.children[1].style.display="none";
    ...
}
```

Checking for matches
```js
// index.js
function checkForMatch(){
    //if matched
    if(flippedCards[0].children[0].src===flippedCards[1].children[0].src) // Checking if the flipped cards have same src i.e are matching
    {        
        ...
    }
    //if not matched
    else
    {
        flippedCards.forEach(flippedCard=>{
            flippedCard.children[0].src="#"; // Removing image src so that it isn't visible through HTML
            flippedCard.children[0].alt="card front face"; // Removing image alt so that it isn't visible through HTML
            flippedCard.children[1].style.display="block";
        })
        ...
    }
    ...
}
```

### CODE REVIEW

**Comment by dev-lovedeep** [comment]
I have closed issue #3 as you suggested. Just incorporate the random functionality we talked about, because from your current code a person can see the imgSrc array in easily win

**Comment by DNA5769** [comment]
> Just incorporate the random functionality we talked about, because from your current code a person can see the imgSrc array and easily win

Done Sir

File: index.js [hunk 1]
```diff
@@ -1,81 +1,123 @@
...
+        if (card.children[0].src!==window.location.href+"#")
+            return;
 
-    //when we have filled two cards check for the match
-    if(flippedCards.length === 2)
-    {
-        setTimeout(checkForMatch,0); 
+        flippedCards.push(card)
+        card.children[0].src=cardImageSrcs[i]; // Setting image source for flipped front face
+        card.children[0].alt=cardImageSrcs[i].split('/').slice(-1)[0].slice(0, -4); // Setting image file name as alt text for flipped front face
```
Comment (dev-lovedeep) [inline index.js:108]: this works fine for '.png' but will not be accurate for other formats. Although doesn't matter much , but just for the sake of completeness.

**Review by dev-lovedeep — COMMENTED** [review]
good idea of using the anonymus function :smiley: . fix this last issue, then I will merge this pr

File: index.js (hunk 1 above)
Comment (DNA5769) [reply index.js]: Done Sir, the code now removes the extension from the image file name and shows the rest as alt text

**Pull request merged**